backstory/src/demo/webapp/vision-demo.js

```diff
@@ -0,0 +1,163 @@
+ * features: fetch Blobstore URL, retrieve/format images and labels, 
+ * display correct file upload
+ */
+
+ /* exported fetchBlobstoreUrl getAnalyzedImages updateFileName */
+
+// FETCH BLOBSTORE URL
+
+/**
+ * Fetches the URL for uploading to Blobstore and adds it to the image upload form.
+ */
+function fetchBlobstoreUrl() {
+  fetch('/blobstore-upload-url')
+      .then((response) => {
+        return response.text();
+      })
+      .then((imageUploadUrl) => {
+        const imageUploadForm = document.getElementById('image-upload-form');
+        imageUploadForm.action = imageUploadUrl;
+      });
+}
+
+// RETRIEVE/FORMAT IMAGES & LABELS
+
+/** Adds the analyzed images to the image-list unordered list element */
+function getAnalyzedImages() {
+  fetch('/analyzed-images')
+      .then((response) => response.json())
+      .then((analyzedImagesObject) => {
+        const imageListElement = document.getElementById('image-list');
+        imageListElement.innerHTML = '';
+
+        for (let i = 0; i < analyzedImagesObject.length; i++) {   
+          const imageUrl = analyzedImagesObject[i].imageUrl;
+          const labelsJsonArray = analyzedImagesObject[i].labelsJsonArray;
+          
+          // parse the labels json and pass it into the formatting array
+          const labelsObj = JSON.parse(labelsJsonArray);
+          const labels = labelsObj.labels;
+
+          imageListElement.appendChild(
+              createRow(imageUrl, labels));
+        }
+      });
+}
+
+/** 
+ * Formats the image url and labels array into HTML
+ *
+ * @param {string} imageUrl - the url of the analyzed image
+ * @param {object} labels - an array of labels & their other properties
+ * @return {object} - an Element containing the analyzed image with its labels and scores 
+ */
+function createRow(imageUrl, labels) {
+  const row = document.createElement('div');
+  row.classList.add('row');
+  row.classList.add('horizontal-flex-container');
+
+  const imageDiv = document.createElement('div');
+  imageDiv.classList.add('image-div');
+
+  // create a div to hold the image for this 
+  const image = document.createElement('img');
+  image.src = imageUrl;
+  image.classList.add('image');
+
+  // add image to its div, and add this div to container (row)
+  imageDiv.appendChild(image);
+  row.appendChild(imageDiv);
+
+  // create divs to hold labels
+  const labelsDiv = document.createElement('div');
+  labelsDiv.classList.add('labels-div');
+  const labelsTable = formatLabelsAsTable(labels);
+  labelsTable.id = 'labels-table';
+  
+  // add labels to row
+  labelsDiv.appendChild(labelsTable);
+  row.appendChild(labelsDiv);
+
+  return row;
+}
+
+/**
+ * Format labels as an HTML table.
+ * 
+ * @param {object} labels - an array of labels (as JSON)
+ * @return {object} - a table representing the labels array
+ */
+function formatLabelsAsTable(labels) {
+  const table = document.createElement('table');
+  table.classList.add('labels-table');
+
+  let index = 0;
```

Comment:
`index` works within a small scope like a for loop (it's clear that it's the incrementer index), but at this scope I'd suggest something more meaningful. Maybe `rowNumber`?

Reply:
That would definitely clarify this variable's purpose. I'll rename it to `rowNumber`.

Reply:
Done